src/utils/createGenerateClassName.js

```diff
@@ -15,5 +21,23 @@ const jssCounter = globalRef[namespace]++
 export default (): generateClassName => {
   let ruleCounter = 0
 
-  return (rule: Rule): string => `${rule.key}-${jssCounter}-${ruleCounter++}`
+  return (rule: Rule, sheet?: StyleSheet): string => {
+    ruleCounter += 1
+
+    if (ruleCounter > maxRules) {
+      warning(
+        false,
+        'You might have a memory leak. Rule counter is at %s.',
+        ruleCounter
+      )
+    }
+
+    if (env === 'production') {
+      return `c${jssCounter}${ruleCounter}`
```

Comment:
@thangngoc89 Here is the `createGenerateClassName` version Material-UI will most likely use in the future.
I'm closing https://github.com/callemall/material-ui/pull/7650. Please defend you use case here :).